projects/canopy/src/lib/paginator/paginator.component.ts

```diff
@@ -0,0 +1,272 @@
+import {
+  ChangeDetectionStrategy,
+  ChangeDetectorRef,
+  Component,
+  EventEmitter,
+  HostBinding,
+  Inject,
+  InjectionToken,
+  Input,
+  Optional,
+  Output,
+  ViewEncapsulation,
+} from '@angular/core';
+
+import {
+  lgIconChevronLeft,
+  lgIconChevronRight,
+  lgIconGotoFirst,
+  lgIconGotoLast
+} from '../icon/icons.interface';
+
+/**
+ * Event raised when the page index is updated
+ */
+export interface PaginationEvent {
+  /**
+   * The selected page index.
+   */
+  pageIndex: number;
+
+  /**
+   * The selected page size.
+   */
+  pageSize: number;
+
+  /**
+   * The previously selected page index.
+   */
+  previousIndex: number;
+}
+
+/**
+ * New paginator default options.
+ */
+export interface PaginationOptions {
+  /**
+   * The page size.
+   */
+  pageSize?: number;
+
+  /**
+   * The list of page sizes.
+   */
+  pageSizes?: Array<number>;
+
+  /**
+   * Flag to hide the page size selector.
+   */
+  hidePageSize?: boolean;
+
+  /**
+   * Flag to show the first and last page buttons.
+   */
+  showFirstLastButtons?: boolean;
+}
+
+/**
+ * Injection token for setting default paginator options.
+ */
+export const LG_PAGINATOR_DEFAULT_OPTIONS = new InjectionToken<PaginationOptions>(
```

Comment:
I'm not familiar with this approach of providing default options, so forgive me if I'm missing something - but it's not a pattern we've used in Canopy so far. Just wondering if it's something we need, or is just part of the component patterns in @lgim/canopy that's made its way over. If it's a something we'd like in Canopy, then perhaps we could consider it as an enhancement to the other components? Again, forgive me if I'm missing something obvious.  Happy to go with the consensus on this 👍

Reply:
It's pretty common approach, you can see https://material.angular.io/components/paginator/api#MAT_PAGINATOR_DEFAULT_OPTIONS for example in Material. It just gives the opportunity to provide default settings for all paginators in an application but isn't required.

Reply:
Thanks, it looks good. I guess I just haven't come across it yet. In my view, we should probably check in with the rest of the team, and see if it's a pattern we want to introduce to Canopy. We tend to be very keen on consistency across all components, so if we adopt it then I imagine we'd look to roll it out to other components too. 

It might be worth doing if it's an approach used in other libraries and something that future adopters of Canopy would find useful.

@thatguynamedandy I know you did a lot of research into Material in the early days of building Canopy, is this a pattern you thought about adopting at the time?

Reply:
I think we used it for the feature toggle service. It's quite a common pattern so I'm personally happy to keep it.

Reply:
Just thinking a bit more about this - maybe it would be better for this to be an input on the component in this case? Just wondering if maybe you have multiple references to that component in the same container module and you want two different default values the input might give a bit more flexibility.

Reply:
There's inputs for each setting so you can override any defaults by using those or if you wanted you could provide the defaults at component level but seems like an unlikely use case?